### PR TITLE
omazureeventhubs: Corrected handling of transport closed failures

### PR DESCRIPTION
**Pull request by alorbach**
- Added test for connection interrupts (requires root)
- Corrected handling of PN_TRANSPORT_CLOSED.
- Make sure Connection is being reestablished trough tryResume
- Enhanced Debug log output

closes: https://github.com/rsyslog/rsyslog/issues/5269

### CODE REVIEW

File: plugins/omazureeventhubs/omazureeventhubs.c [hunk 1]
```diff
@@ -712,12 +747,30 @@ CODESTARTcommitTransaction
 			}
 		}
 
-		if (	pWrkrData->bIsConnected == 1 &&
-			iNeedSubmission > 0 &&
-			pn_link_credit(pWrkrData->pnSender) > 0) {
-			pn_connection_wake(pWrkrData->pnConn);
-			/* Grant enough credit to bring it up to BATCH: */
-			// pn_link_flow(pWrkrData->pnSender, iToSubmit);
```
Comment (rgerhards) [inline plugins/omazureeventhubs/omazureeventhubs.c:720]: pls remove commented-out code

File: plugins/omazureeventhubs/omazureeventhubs.c [hunk 2]
```diff
@@ -712,12 +747,30 @@ CODESTARTcommitTransaction
...
+			} else {
+				DBGPRINTF("omazureeventhubs[%p]: commitTransaction Suspended=%s Connecting=%s\n",
+					pWrkrData,
+					pWrkrData->bIsSuspended == 1 ? "YES" : "NO",
+					pWrkrData->bIsConnecting == 1 ? "YES" : "NO");
+				if (pWrkrData->bIsSuspended == 1 && pWrkrData->bIsConnecting == 0) {
+					ABORT_FINALIZE(RS_RET_SUSPENDED);
+				}
+				// pn_connection_wake(pWrkrData->pnConn);
+				// ABORT_FINALIZE(RS_RET_SUSPENDED);
```
Comment (rgerhards) [inline plugins/omazureeventhubs/omazureeventhubs.c]: pls remove commented-out code

**Review by rgerhards — CHANGES_REQUESTED** [review]

File: plugins/omazureeventhubs/omazureeventhubs.c (hunk 1 above)
Comment (alorbach) [reply plugins/omazureeventhubs/omazureeventhubs.c:720]: Sure, removed that commented code.

File: plugins/omazureeventhubs/omazureeventhubs.c (hunk 2 above)
Comment (alorbach) [reply plugins/omazureeventhubs/omazureeventhubs.c]: removed that as well

File: plugins/omazureeventhubs/omazureeventhubs.c (hunk 1 above)
Comment (rgerhards) [reply plugins/omazureeventhubs/omazureeventhubs.c:720]: I still see it?

Comment (alorbach) [reply plugins/omazureeventhubs/omazureeventhubs.c:720]: I see it removed in the force push compare: 
https://github.com/rsyslog/rsyslog/compare/4f01cf92ce5c1927bb8412c9637ce79a83914fed..eece807639fa8e3df62808394f847792417f6cf9

Comment (rgerhards) [reply plugins/omazureeventhubs/omazureeventhubs.c:720]: sry, my fault. this is **removed** code...

**Review by rgerhards — APPROVED** [review]

**Comment by alorbach** [comment]
Can be merged

**Pull request merged**